client/components/data/query-whois/index.jsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingWhois } from 'state/selectors';
+import { requestWhois } from 'state/domains/management/actions';
+
+class QueryWhois extends Component {
+	componentWillMount() {
+		if ( this.props.requesting ) {
+			return;
+		}
+		this.props.requestWhois( this.props.domain );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QueryWhois.propTypes = {
+	domain: PropTypes.string,
+	requesting: PropTypes.bool,
+	requestWhois: PropTypes.func
+};
+
+export default connect(
+	( state, { domain } ) => ( { requesting: isRequestingWhois( state, domain ) } ),
```

Comment:
So I've read up and down your PRs (both the one above and your actual ESlint PR).. I think I'm getting the general point but I'm struggling to understand what that translates to in `connect` code. How would you write this specific one? Something like this?

```
export default connect(
	( state, { domain } ) => {
		const isRequesting = isRequestingWhois( state, domain );
		return { isRequesting };
	},
	{ requestWhois }
)( QueryWhois );
```